sr_data_visualization/src/sr_data_visualization/data_plot.py

```diff
@@ -125,6 +139,28 @@ def create_traces(self):
                        Trace("Effort", Qt.blue, self.x_data),
                        Trace("Velocity", Qt.green, self.x_data)]
 
+    def plot_data(self, plot, side, new_sub=True):
+        self.joint_name = self.joint_name.replace("lh", side).replace("rh", side)
+
+        self.clear_data()
+
+        if new_sub:
```

Comment:
what is new_sub? we don' plot if it's not True? is there a reason why in line 94 this is not the same?

Reply:
new_sub is contained in the plot_data for the Joint States and Motor Stats because they don't need to change subscription when the side is changed (/joint_states and /diagnostic_agg contains the info for both sides). So since in the proposed changes plot_data would also handle changing side, when the combobox of the side is modified, there should be a way to inform to change the side without having to create a new subscription. That's why [here](https://github.com/shadow-robot/sr-visualization/blob/4325d370839b3b5810c767fda22f175c4a63cf9e/sr_data_visualization/src/sr_data_visualization/sr_data_visualizer_gui.py#L196) plot_data is called with new_sub=False. If you think of a cleaner way of doing this, let me know.